src/Extensions/src/App.Metrics.Extensions.Hosting/MetricsReporterBackgroundService.cs

```diff
@@ -97,49 +103,50 @@ private async Task ExecuteOnceAsync(CancellationToken cancellationToken)
 
                 flushTask.Increment();
 
-                await taskFactory.StartNew(
```

Comment:
Why was this necessary before? cc @alhardy

Reply:
The design of the BackgroundService was taken from the example [here](https://blog.maartenballiauw.be/post/2017/08/01/building-a-scheduled-cache-updater-in-aspnet-core-2.html)

"So TaskFactory.StartNew()… Why not simply await them here, you ask? Well, what if you schedule a task to run every minute but that task never returns (or returns after a couple of minutes)? Our scheduler would be useless. So instead we’re spawning tasks outside of our scheduler so at least it can keep its promises"

Reply:
So we need to add that back in this PR then?

Reply:
What was the issue in spawning a new task and using the application lifetime token as the cancellation token?
Just thinking if there was some delay in flushing metrics between intervals some metrics would be lost right? I.e. serialized metrics, including the timestamp, could be pending on each tasks if there is for whatever reason delays/throttling on the tsdb.

Reply:
Basically when the host is shutting down and I call the function with `.Wait()` it will then queue this task but the apphost is waiting on all existing Tasks but this new one will basically hit the first await and then never resume. The apphost exits and this just disappears, not even cancelled.

I could add it back and then add a parameter which essentially will optionally do it but it just can't work to start a new task while shutting down.

Reply:
OK, let's leave the PR as is.